dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -324,23 +325,40 @@ private void validateIncomingParams(final Map<String, Param> incomingParams, fin
         for (final Entry<String, Param> incomingParamEntry : incomingParams.entrySet()) {
             final String incomingParamName = incomingParamEntry.getKey();
             final Param describedParam = serviceDescriptorParams.get(incomingParamName);
+            final Param incomingParam = incomingParamEntry.getValue();
+
             if(serviceDescriptor.isAllowExtraParameters() && null == describedParam){
                //if the param isn't found in our description but the allow extra params flag is true we're ok
+               //Validate type..
                continue;
             }
+
             //If the flag isn't true. Then we must reject the unknown param.
             if(null == describedParam) {
                 throw new IllegalArgumentException(String.format(
                         "Params named `%s` can not be matched against service descriptor. ",
                         incomingParamName));
             }
 
-            final Param incomingParam = incomingParamEntry.getValue();
             //We revise the incoming param against the definition loaded from the yml.
             if(describedParam.isRequired() && UtilMethods.isNotSet(incomingParam.getValue())){
                throw new IllegalArgumentException(
                String.format("Params named `%s` is marked as required in the descriptor but does not have any value.", incomingParamName));
             }
+
+
+
+            if(Type.BOOL.equals(describedParam.getType())){
+                if(UtilMethods.isSet(incomingParam.getValue())){
+                    if(!("true".equalsIgnoreCase(incomingParam.getValue()) || "false".equalsIgnoreCase(incomingParam.getValue()))){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4979325)